src/FSFile.cpp

```diff
@@ -0,0 +1,38 @@
+#include "FSFile.h"
+//Set some arbitrary values
+FSFile::FSFile(std::string path):FSItem(path){
+	this->finfo->st->st_mode = S_IFREG | 0644;
+	this->finfo->st->st_nlink = 1;
+	this->finfo->st->st_size=0;
+	this->finfo->st->st_uid=getuid();
+	this->finfo->st->st_gid=getgid();
+	this->finfo->st->st_atime = time( NULL );
+	this->finfo->st->st_mtime = time( NULL );
+	
+}
+FSFile::FSFile(std::string path, std::string contents):FSFile(path){
+	this->fileContents=std::string(contents);
+	this->finfo->st->st_size=contents.length();
+}
+std::shared_ptr<std::string> FSFile::getFileContents(){
+	return this->getFileContents(0);
+
+}
+std::shared_ptr<std::string> FSFile::getFileContents(int offset){
+	this->finfo->st->st_atime = time( NULL );
+	try{
+		auto str = std::make_shared<std::string>(this->finfo->st->st_size-1, 'x');
+		return str;
+	} catch(std::exception &ex){
+		return nullptr;
+		
+	}
+	
```

Comment:
What kind of exception is the try-catch block protecting against? If it's only to protect against out of memory issues, the program will crash anyway...

Is returning the file contents using shared_ptr most effective? The problem is that currently this implementation achieves about 700KB/s during sequential read - the expected throughput is about 1,000 time higher. Can we skip the construction of std::string altogether and directly set the output buffer provided by fuse in read callback?